File: Solutions/Medium/1578. Minimum Time to Make Rope Colorful/Solution.py
class Solution:
    def minCost(self, colours: str, neededTime: List[int]) -> int:
        n = len(colours)
        prev = "-"
        lastNum = 0
        result = 0
        for i in range(n):
            if (prev != colours[i]):
                prev = colours[i]
                lastNum = neededTime[i]
            elif (lastNum >= neededTime[i]):
                result += neededTime[i]
            else:
                result += lastNum
                lastNum = neededTime[i]
        return result

File: Solutions/Medium/1578. Minimum Time to Make Rope Colorful/Solution.cs
public class Solution {
    public int MinCost(string colours, int[] neededTime) {
        int n = colours.Length;
        char prev = '-';
        int removeCost = 0, result = 0;
        for(int i = 0; i < n; i++){
            if(prev != colours[i]){
                prev = colours[i];
                removeCost = neededTime[i];
            }
            else if(removeCost >= neededTime[i]){
                result += neededTime[i];
            }
            else{
                result += removeCost;
                removeCost = neededTime[i];
            }
        }
        return result;
    }
}